traitlets/traitlets.py

```diff
@@ -587,6 +588,8 @@ def _validate(self, obj, value):
             return value
         if hasattr(self, 'validate'):
             value = self.validate(obj, value)
+        for validator in self._allows:
+            value = validator(self, value)
```

Comment:
Leave out the `value = ` part to restrict `allows` to validation and prevent issues related to the ordering of the validators.